src/WebView.android.tsx

```diff
@@ -119,12 +129,26 @@ const WebViewComponent = forwardRef<{}, AndroidWebViewProps>(({
   }), [setViewState, webViewRef]);
 
   const directEventCallbacks = useMemo(() => ({
-    onShouldStartLoadWithRequest,
-    onMessage,
-  }), [onMessage, onShouldStartLoadWithRequest]);
+    onShouldStartLoadWithRequest: (
+      event: ShouldStartLoadRequestEvent & { messagingModuleName?: string },
+    ) => {
+      if (event.messagingModuleName === messagingModuleName) {
+        const { messagingModuleName: _, ...rest } = event;
+        onShouldStartLoadWithRequest(rest);
+      }
+    },
+    onMessage: (
+      event: WebViewMessageEvent & { messagingModuleName?: string },
+    ) => {
+      if (event.messagingModuleName === messagingModuleName) {
+        const { messagingModuleName: _, ...rest } = event;
+        onMessage(rest);
+      }
+    },
+  }), [messagingModuleName, onMessage, onShouldStartLoadWithRequest]);
 
   useEffect(() => {
-    BatchedBridge.registerCallableModule(messagingModuleName, directEventCallbacks);
+    registerCallableModule('RNCWebViewMessagingModule', directEventCallbacks);
```

Comment:
Should this first parameter be static and not `messagingModuleName`? Perhaps this is what may cause #3368